operator/src/main/java/org/projectnessie/operator/resource/NessieSpec.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.operator.resource;
+
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+
+public class NessieSpec {
+
+  public enum VersionStoreType {
+    IN_MEMORY,
+    ROCKSDB,
+    DYNAMODB,
+    MONGODB,
+    CASSANDRA,
+    JDBC,
+    BIGTABLE
+  }
+
+  @JsonPropertyDescription("The number of replicas to run.")
+  private int size = 1;
+
+  @JsonPropertyDescription("The image to use for the Nessie server.")
+  private NessieImage image = new NessieImage();
+
+  @JsonPropertyDescription("The log level to use for the Nessie server.")
+  private String logLevel = "INFO";
+
+  @JsonPropertyDescription("The type of version store to use.")
+  private VersionStoreType versionStoreType = VersionStoreType.IN_MEMORY;
```

Comment:
I wonder whether we need to add all possible config options here or somehow leverage/generate the `helm/` stuff? Looks like there would be two places to maintain things.

Reply:
Unfortunately, I think we are in for maintaining both deployment methods separately. I don't know of any tool that can generate an operator from a Helm chart of vice-versa.

(Note: the operator project does generate Helm charts for deploying the operator itself. These are distinct from the Helm charts to deploy Nessie.)